### ISSUE TITLE
Extra hints for the window constructor

### ISSUE THREAD

**Issue by rgiot** [phaazon/luminance-glfw#3]
Hello,

`GlfwSurface::new` makes some `glfw.window_hint` calls. However it is not possible for the users of this lib to specify other hints to do (e.g. `window_hint(glfw::WindowHint::Samples(Some(4)))`) or to change value of some already set (e.g. using another version of opengl).

As this constructor is tightly linked to `luminance::Surface::new` I guess it is not possible to pass thes information as an argument.

Is there a plan in the future to allow finder control to the user of the lib ?
If no, it is not really a problem as re-implenting a specific GlwSurface would takes no more that one hundred lines

**Comment by phaazon** [comment]
For now yeah, but it’s _very_ easy to let users provide them and the design of the overall thing encourages such additions.

Which kind of items do you think of?

**Comment by rgiot** [comment]
When writing this post I was mainly thinking about antialising stuff. But there is also the OpenGL version choice (which is also harcoded in luminance).

**Comment by phaazon** [comment]
Yeah, it has to be hardcoded, since it would make the `luminance` break if you specify a too high / too low setup.

**Comment by phaazon** [comment]
I’ll add the anti-aliasing stuff when I find some spare time. :)
